### ISSUE TITLE
Getting exception when installing Alerta

### ISSUE THREAD

**Issue by jaco-botha** [googley/ansible-alerta#1]
I am trying a test install of Alterta on Ubuntu 16.04.01

The playbook gets stuck on this:

TASK [googley.alerta : Install Alerta.] ****************************************
task path: /etc/ansible/roles/googley.alerta/tasks/main.yml:10
ESTABLISH LOCAL CONNECTION FOR USER: root
localhost EXEC ( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553 )" )
localhost PUT /tmp/tmpznaFX4 TO /home/jaco/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553/pip
localhost EXEC LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/jaco/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553/pip; rm -rf "/home/jaco/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553/" > /dev/null 2>&1
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/home/jaco/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553/pip", line 2329, in <module>
    main()
  File "/home/jaco/.ansible/tmp/ansible-tmp-1484387067.58-30566960144553/pip", line 326, in main
    has_vcs = bool(name and re.match(r'(svn|git|hg|bzr)\+', name))
  File "/usr/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "pip"}, "parsed": false}

I am not familiar with debugging ansible modules, any ideas or pointers would be appreciated.

**Comment by ElieDeloumeau** [comment]
Is this the -vvvv debugging output?

**Comment by jaco-botha** [comment]
That was the output with -vvv

I reran with -vvvv but I don't see any more details ?:

TASK [googley.alerta : Install Alerta.] ****************************************
task path: /etc/ansible/roles/googley.alerta/tasks/main.yml:10
ESTABLISH LOCAL CONNECTION FOR USER: root
localhost EXEC ( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893 )" )
localhost PUT /tmp/tmpXz7XH4 TO /home/jaco/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893/pip
localhost EXEC LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/jaco/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893/pip; rm -rf "/home/jaco/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893/" > /dev/null 2>&1
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/home/jaco/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893/pip", line 2329, in <module>
    main()
  File "/home/jaco/.ansible/tmp/ansible-tmp-1484514013.52-125669059908893/pip", line 326, in main
    has_vcs = bool(name and re.match(r'(svn|git|hg|bzr)\+', name))
  File "/usr/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "pip"}, "parsed": false}

PLAY RECAP *********************************************************************
localhost                  : ok=22   changed=0    unreachable=0    failed=1

**Comment by jaco-botha** [comment]
I have found a workaround.

It seems when pip is invoked like this (in main.yml ):
---snip---
```
- name: Install Alerta.
  become: true
  pip:
    name: 
        - alerta
        - alerta-server
    virtualenv: '{{ alerta_dir }}'
```
---snip---
We get this failure. 

If I change the playbook to the following I don't get this error any more:
--snip---
```
- name: Install Alerta.
  become: true
  pip:
    name: alerta
    virtualenv: '{{ alerta_dir }}'

- name: Install Alerta server.
  become: true
  pip:
    name: alerta-server
    virtualenv: '{{ alerta_dir }}'
```
---snip---

**Comment by ElieDeloumeau** [comment]
Ok, with which version of ansible?

**Comment by jaco-botha** [comment]
ansible 2.0.0.2

I am running into the next hurdle, which seems to be that the apache.yml install uses include_role which is only available in ansible 2.2. Will update to ansible 2.2 now to see if I can finally complete the install :)

**Comment by jaco-botha** [comment]
With 2.2 the install runs through without a hitch. I reverted my changes to main.yml as well and the original yml no longer throws an exception.
